### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by LionSnarl**
Will update to provide Stretch challenges

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -6,22 +6,84 @@ var secretCodeWord1 = "lackadaisical"
...
+//return that value is equal to 0
+//  else return my value (value 'as is')
+    if(value === "a"){
+      return value = 4
+    }else if (value === "e"){
+      return value = 3
+    }else if (value === "i"){
+      return value = 1
+    }else if (value === "o"){
+      return value = 0
```
Comment (minachuong) [inline code-challenges.js:34]: consider a data structure to store the mapping from letter to number

File: code-challenges.js [hunk 2]
```diff
@@ -31,9 +93,33 @@ var hand3 = [5, 5, 5, 5, 4]
...
+//there are only two possible "winning" scenarios if your logic in comparing only 5 "cards"
+//if numbers start 0-4, if 0==1==2, and 3==4 or else if 0==1 and 2==3==4
+//true (You have a full house)
+//anything else would be false
+//A function taking an array of 5 numbers
+ const fullHouse = (array) => {
+//.sort least to greatest
+   let sortArray = array.sort((a,b)=>a-b)
+//if index 0 equals index 1, index 2.....
+   if (sortArray[0] === sortArray[1] && sortArray[0] === sortArray[2] && sortArray[3] === sortArray[4]){
```
Comment (minachuong) [inline code-challenges.js:108]: Consider using a data structure that models a full house. This explicit check does the trick but it's not obvious to someone else what this conditional is trying to do.

File: interview-questions.md [hunk 3]
```diff
@@ -7,65 +7,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database is a form of database that holds information in tables consisting of rows and columns. The information is stored to show relationships between sets of data inside the rows and columns. There are many different types of databases, some do not follow how the data is related to each other and they do not have to consist of rows and columns.
 
-  Researched answer:
+  Researched answer: Relational databases are a type of database that throw pieces of data that are related to each other onto a table designed with columns and rows. There are many kinds of databases, and you can build them to suit your needs.
```
Comment (minachuong) [inline interview-questions.md:28]: yup. Main thing in web development is that there are relational and non-relation databases. It'd be good to know the difference between these two.

File: interview-questions.md [hunk 4]
```diff
@@ -7,65 +7,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 7. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are sort of like ID tags that are used to help identify every row of data. So they are sort of like unique identifiers that help pinpoint exact data sets inside a database. They are important because there could be many first and last names in a data base, but they are all associated with a single primary key to refer to.
 
-  Researched answer:
+  Researched answer: Primary keys are important because they are unique identifiers designated to identify each table record. They are important because sets of data can be assigned a single row and all pieces of data that are related to each other can all be identified by a primary key. They are like social security numbers in the United States, so with my name Ryan Sarll, or with another name like Jose Gonzales, there is probably more than one person in the US named our names, so we are given SSN's, or Primary Keys, to refer to us as individuals that contain all our relevant data.
```
Comment (minachuong) [inline interview-questions.md:60]: I really like that you provided an example of a primary key and the significance for how it's used! Keep this up!

File: ruby_challenges.rb [hunk 5]
```diff
@@ -1,27 +1,71 @@
 # ASSESSMENT 5: Ruby Coding Practical Questions
-
+# MINASWAN :]
```
Comment (minachuong) [inline ruby_challenges.rb:2]: Matz fan?! 😄 Very cool

File: ruby_challenges.rb [hunk 6]
```diff
@@ -1,27 +1,71 @@
...
+  @frame_size = '168cm'
+  @bell = 'Ding Ding!'
+  @speed = 0
+  end
 
 # Expected output example: 'The Trek bike has 2 wheels and a 168cm frame.'
-
-
+def get_info
+  puts "The #{model} bike has #{wheels} wheels and a #{frame_size} frame."
```
Comment (minachuong) [inline ruby_challenges.rb:17]: Cool use of class methods. Consider why and when you'd want to use getter methods versus instance variables.

File: code-challenges.js (hunk 1 above)
Comment (minachuong) [reply code-challenges.js:34]: Also know that if you're using a `return` in the if-blocks, you don't need to assign the number to the `value`. You can literally return the number (e.g.`return 4`) and it'll continue to iterating onto the next term.

**Review by minachuong — APPROVED** [review]
Great job. Really good use of examples in your interview questions. Consider using key-value pairs to store data to reduce if statements.

File: ruby_challenges.rb (hunk 5 above)
Comment (LionSnarl) [reply ruby_challenges.rb:2]: Ruby is fun :)

**Pull request merged**